src/core/thread/mle_router.cpp

```diff
@@ -1108,20 +1108,32 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
                      leaderData.GetWeighting(), peerParitionId,
                      mLeaderData.GetWeighting(), mLeaderData.GetPartitionId());
 
-        if ((leaderData.GetWeighting() > mLeaderData.GetWeighting()) ||
-            (leaderData.GetWeighting() == mLeaderData.GetWeighting() &&
-             peerParitionId > mLeaderData.GetPartitionId()))
+        if (mDeviceMode & ModeTlv::kModeFFD)
         {
-            otLogDebgMle("trying to migrate\n");
-            BecomeChild(kMleAttachBetterPartition);
+            if ((leaderData.GetWeighting() > mLeaderData.GetWeighting()) ||
+                (leaderData.GetWeighting() == mLeaderData.GetWeighting() &&
+                 peerParitionId > mLeaderData.GetPartitionId()))
+            {
+                otLogDebgMle("trying to migrate\n");
+                BecomeChild(kMleAttachBetterPartition);
```

Comment:
A device should not attempt to reattach if the advertisement came from it's current parent.